### PR TITLE
MSVC fixes

### PR DESCRIPTION
**Pull request by bjornpiltz**

### CODE REVIEW

File: src/base/reconstruction.cc [hunk 1]
```diff
@@ -567,7 +567,6 @@ std::vector<image_t> Reconstruction::FilterImages(
     const class Image& image = Image(image_id);
     const class Camera& camera = Camera(image.CameraId());
     if (image.NumPoints3D() == 0) {
-      DeRegisterImage(image_id);
```
Comment (ahojnnes) [inline src/base/reconstruction.cc]: Why is this?

File: src/ext/FLANN/algorithms/kmeans_index.h [hunk 2]
```diff
@@ -373,7 +373,7 @@ class KMeansIndex : public NNIndex<Distance>
         /**
          * The cluster center.
          */
-        DistanceType* pivot;
+        DistanceType* pivot=nullptr;
```
Comment (ahojnnes) [inline src/ext/FLANN/algorithms/kmeans_index.h]: One of the many uninitialized variables in FLANN causing problems :-)

File: src/mvs/CMakeLists.txt [hunk 3]
```diff
@@ -20,6 +20,10 @@ if(CUDA_ENABLED)
         # Workaround for Visual Studio compiler error,
         # due problems with compiling too complex code.
         string(REPLACE "/Zo" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+
+        # Avoid pulling in too many header files through <windows.h>
+        add_definitions(-DWIN32_LEAN_AND_MEAN)
```
Comment (ahojnnes) [inline src/mvs/CMakeLists.txt]: Under which MSVC version is this a problem? Just out of curiosity.

**Comment by ahojnnes** [comment]
Thanks very much for the fixes!

File: src/mvs/CMakeLists.txt (hunk 3 above)
Comment (bjornpiltz) [inline src/mvs/CMakeLists.txt]: I have only got CUDA to work with MSVC 2013, so it only shows up there. It probably also depends on which Win SDK you have installed.

**Pull request merged**